scripting/srccoop_addon_effects_fix.sp

```diff
@@ -0,0 +1,192 @@
+#include <clientprefs>
+#include <sdkhooks>
+#include <sourcemod>
+#include <srccoop_api>
+
+#pragma newdecls required
+#pragma semicolon 1
+#pragma tabsize 0
+
+public Plugin myinfo =
+{
+	name        = "SourceCoop damage effects fix",
+	author      = "Jimbab",
+	description = "Client side damage effects fix for all weapons.",
+	version     = "1.1.0",
+	url         = "https://github.com/ampreeT/SourceCoop"
+};
+
+public void OnPluginStart()
+{
+	InitSourceCoopAddon();
+}
+
+public void OnEntityCreated(int entityIndex, const char[] szClassname)
+{
+	if (CBaseEntity(entityIndex).IsClassNPC())
+		SDKHook(entityIndex, SDKHook_OnTakeDamage, DispatchEffects);
```

Comment:
Use SDKHook_OnTakeDamagePost hook in case other plugins block or modify the event.

Reply:
I see, alright. Just so I understand correctly; relative to `OnTakeDamage`, is `OnTakeDamagePost` fired the following frame? I am assuming so.

Reply:
No, same frame, right after the pre hook.

Reply:
Oh right.. hmm. Okay then.

Anyways, OnTakeDamagePost doesn't work for whatever reason. It passes different and often invalid values for what I was using it for... Also, I had been, and was still investigating EffectDispatch, and it doesn't seem very promising. None of the values the tempent holds are consistently unique between effects.

So, unless my knowledge of what's possible is limiting me here, I think EffectDispatch is a no-goer.